### PR TITLE
feat: linear imputation in Resample

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY
Adding a new `resample method`: linear imputation

### TESTING INSTRUCTIONS

1. download [DailyDelhiClimateTrain.csv](https://www.kaggle.com/datasets/sumanthvrao/daily-climate-time-series-data) and create a new dataset in your local. (Use PostgreSQL as current testing)
2. open explore page and select DailyDelhiClimateTrain dataset
3. change viz type to `Bar Chart`, notice that you should pick up echart version
4. drag date in X-Axis
5. set time range from `2014-12-10` to `2014-12-20`
6. set metric to `MAX(meantemp)::INTEGER`
7. set filter to `date not in ('2014-12-11'::TIMESTAMP, '2014-12-12'::TIMESTAMP)`
8. you get a bar chart without data for `12-11 and 12-12`.
image

9. select `1 calendar day frequency` and `Linear imputation` in Advanced analytics panel, then click `Run`. 
image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-chart-controls/src/sections/advancedAnalytics.tsx [hunk 1]
```diff
@@ -170,6 +170,7 @@ export const advancedAnalyticsControls: ControlPanelSectionConfig = {
           choices: [
             ['asfreq', 'Null imputation'],
             ['zerofill', 'Zero imputation'],
+            ['linear', 'Linear imputation'],
```
Comment (villebro) [inline superset-frontend/packages/superset-ui-chart-controls/src/sections/advancedAnalytics.tsx:173]: nit: 
```suggestion
            ['linear', 'Linear interpolation'],
```

**Review by villebro — COMMENTED** [review]
Niiiice! Two minor nits, other than that really looking forward to getting this merged 👍

File: superset/utils/pandas_postprocessing/utils.py [hunk 2]
```diff
@@ -92,6 +92,10 @@
     "P1W/1970-01-04T00:00:00Z": "W",
 }
 
+RESAMPLE_METHOD = tuple(
+    ["asfreq", "bfill", "ffill", "linear", "median", "mean", "sum",]
+)
```
[the remark this answers is not in the document]
Comment (zhaoyongjie) [reply superset/utils/pandas_postprocessing/utils.py:97]: Usually, I force myself to use the explicit expression, so eliminate `foo = ("bar")` error.  Thanks for those catches. I will update.

**Review by villebro — APPROVED** [review]
LGTM

**Pull request merged**